packages/binding-coap/src/coap-client.ts

```diff
@@ -214,31 +214,7 @@ export default class CoapClient implements ProtocolClient {
     private generateRequest(form: CoapForm, dflt: string, observable = false): any {
         const options: CoapRequestConfig = this.uriToOptions(form.href);
 
-        options.method = dflt;
-
-        if (typeof form["coap:methodCode"] === "number") {
-            console.debug("[binding-coap]", "CoapClient got Form 'methodCode'", form["coap:methodCode"]);
-            switch (form["coap:methodCode"]) {
-                case 1:
-                    options.method = "GET";
-                    break;
-                case 2:
-                    options.method = "POST";
-                    break;
-                case 3:
-                    options.method = "PUT";
-                    break;
-                case 4:
-                    options.method = "DELETE";
-                    break;
-                default:
-                    console.warn(
-                        "[binding-coap]",
-                        "CoapClient got invalid 'methodCode', using default",
-                        options.method
-                    );
-            }
-        }
+        options.method = form["cov:methodName"] ? form["cov:methodName"] : dflt;
```

Comment:
With this change both the CoAP and the CoAPS clients would be more similar to the HTTP client. Let me know if the debug messages should be kept.

Reply:
I wonder what should happen IF it is not a *known* method. Fail I guess?

It might be useful to extract the current list in coap.ts so that here we can check whether its is part of this list before setting the value.

Reply:
The old approach was to use the default value and print a debug message if the method is undefined or unknown. I am not sure if this approach or raising an error would better.

In any case, I think the solution we agree on should probably also be implemented in other bindings to have a consistent behavior, I guess? At least in the `binding-http`, package such a check does not seem to happen if I am not mistaken.

Reply:
I now reintroduced checks with the default method as a fallback. Let me know if there should be an error thrown instead :)